semgrep_output_v1.atd

```diff
@@ -233,7 +251,7 @@ type core_match_extra <python decorator="dataclass(frozen=True)"> = {
   ?fix: string option;
 
   ?dataflow_trace: match_dataflow_trace option;
-  engine_kind: engine_kind;
+  engine_kind: engine_flavor;
```

Comment:
Same comment that in the other PR. the difference between kind and flavor is confusing.
Use just one, and since the field is called engine_kind, I would use engine_kind for the type.

Reply:
I would like to separate engine_type in general (for rules, for checking the requested engine, etc) from engine_type for findings (which attaches information about which feature was used). Hopefully this rename is cleaner. I was hesitate to rename `engine_kind` in the atd because I'm worried about backwards compatibility and I think this is reasonable. I did rename it in the OCaml code.